lib/charms/grafana_k8s/v0/grafana_dashboard.py

```diff
@@ -1636,29 +1635,60 @@ def _remove_all_dashboards_for_relation(self, relation: Relation) -> None:
             self.on.dashboards_changed.emit()  # pyright: ignore
 
     def _to_external_object(self, relation_id, dashboard):
+        decompressed = LZMABase64.decompress(dashboard["content"])
+        as_dict = json.loads(decompressed)
+
+        dashboard_title = as_dict.get("title", "")
+        dashboard_uid = as_dict.get("uid", "")
+
+        try:
+            dashboard_version = int(as_dict["version"])
+        except (KeyError, ValueError):
+            logger.warning("Dashboard '%s' (uid '%s') is missing a '.version' field or is invalid (must be integer); using '0' as fallback", dashboard_title, dashboard_uid)
+            dashboard_version = 0
```

Comment:
Wouldn't be valuable to know exactly which error are we facing?

```suggestion
        dashboard_version = 0
        try:
            dashboard_version = int(as_dict["version"])
        except KeyError:
            logger.warning("dashboard '%s' (uid '%s') is missing a '.version' field", dashboard_title, dashboard_uid)
        except ValueError:
            logger.warning("dashboard '%s' (uid '%s') '.version' field is invalid (must be integer); using '0' as fallback", dashboard_title, dashboard_uid)
```

Reply:
I started like that, and then realized that the action taken wouldn't change between the two cases: need to inspect and fix the version field.